### PR TITLE
@types/gm Add GetterOptions for working with streams and identify operations 

### PR DESCRIPTION
**Pull request by maartenvanvliet**
First commit on DefinitelyTyped so I hope I've done things correctly.

Added GetterOptions interface because identify operations need to buffer readStream in memory if you need to write()/stream() afterwards and the current definitions did not cover that.
See: https://github.com/aheckmann/gm/blob/c6a6c5a/README.md#streams

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
  - I see some linting errors, fixed a couple but still some remain.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - Documentation: https://github.com/aheckmann/gm/blob/c6a6c5a/README.md#streams
  - Source: https://github.com/aheckmann/gm/blob/c6a6c5a/lib/getters.js#L39
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by andy-ms** [comment]
@maartenvanvliet There are some remaining lint errors.

**Comment by maartenvanvliet** [comment]
@andy-ms I know, they were there before. Most are overloaded functions that should be using union types instead. There are now close to 2000 lint errors and I don't have the time to fix it tonight.

**Comment by maartenvanvliet** [comment]
@andy-ms Fixed it, but disabled the "unified-signatures" exception rule. Too many linting errors for now.

File: types/gm/index.d.ts [hunk 1]
```diff
@@ -258,18 +246,7 @@ declare namespace m {
...
-        /** Ignore aspect ratio */
-        geometry(width: number, height: number, option: '!'): State;
-        /** Width and height are minimum values */
-        geometry(width: number, height: number, option: '^'): State;
-        /** Change dimensions only if image is smaller than width or height */
-        geometry(width: number, height: number, option: '<'): State;
-        /** Change dimensions only if image is larger than width or height */
-        geometry(width: number, height: number, option: '>'): State;
+
         geometry(width: number, height?: number, option?: string): State;
```
Comment (ChaosinaCan) [inline types/gm/index.d.ts]: Should the type of option be a ResizeOption?

**Review by joelspadin — CHANGES_REQUESTED** [review]

File: types/gm/index.d.ts (hunk 1 above)
Comment (maartenvanvliet) [inline types/gm/index.d.ts]: Yes, good call

**Comment by ChaosinaCan** [comment]
👍

**Pull request merged**